src/shapes/Object/StackedObject.ts

```diff
@@ -95,15 +98,22 @@ export class StackedObject<
       return {
         fork: [],
         otherFork: [],
-        common: [this, ...this.getAncestors(strict)],
-      } as AncestryComparison<S>;
+        common: [this, ...(this.getAncestors(strict) as Ancestors<S>)],
+      };
     }
-    const ancestors = this.getAncestors(strict);
-    const otherAncestors = other.getAncestors(strict);
+
+    type AncestorsTuple = Omit<Ancestors<S>, 'slice'> & {
+      // Don't widen the tuple to a generic array
+      slice: (from?: number, to?: number) => Ancestors<S>;
+    };
+    const ancestors = this.getAncestors(strict) as AncestorsTuple;
+    const otherAncestors = other.getAncestors(strict) as AncestorsTuple;
+
     //  if `this` has no ancestors and `this` is top ancestor of `other` we must handle the following case
     if (
       ancestors.length === 0 &&
       otherAncestors.length > 0 &&
+      isGroup(this) &&
```

Comment:
I understand that this line, and line 131 and line 139 are adding the check `isGroup(x)` but in all three cases the group condition is checked implcitly by the statement that follows.
if `this === otherAncestors[otherAncestors.length - 1]` then `this` must be a group, and if `ancestor === other` then other must be a group and etc etc, but i wish those details are called out in the pr description rather than to the reader to understand why there are code changes in a type changes pr.

Reply:
It was added to make TS happy as otherwise it thinks you're comparing an object that is potentially not a group to an ancestor. I had pointed it out in the PR description:

> After I added some artificial isGroup type narrowing TS is a bit happier

Reply:
i read that, i was asking if you think the ancestor could be a non group. i was arguing that is always a group.

Reply:
An ancestor can be a non-group as it can be a canvas or Group, but `this` must be an object, thus it must be a Group as well